.github/workflows/scorecards.yml

```diff
@@ -0,0 +1,59 @@
+name: Scorecards supply-chain security
+on:
+  # Only the default branch is supported.
+  branch_protection_rule:
+  schedule:
+    - cron: '15 4 * * 4'
+  push:
+    branches: [ "master" ]
+
+# Declare default permissions as read only.
+permissions: read-all
+
+jobs:
+  analysis:
+    name: Scorecards analysis
+    runs-on: ubuntu-latest
+    permissions:
+      # Needed to upload the results to code-scanning dashboard.
+      security-events: write
+      # Used to receive a badge.
+      # id-token: write
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846 # tag=v3.0.0
+        with:
+          persist-credentials: false
+
+      - name: "Run analysis"
+        uses: ossf/scorecard-action@865b4092859256271290c77adbd10a43f4779972 # tag=v2.0.3
+        with:
+          results_file: results.sarif
+          results_format: sarif
+          # (Optional) Read-only PAT token. Uncomment the `repo_token` line below if:
+          # - you want to enable the Branch-Protection check on a *public* repository, or
+          # - you are installing Scorecards on a *private* repository
+          # To create the PAT, follow the steps in https://github.com/ossf/scorecard-action#authentication-with-pat.
+          # repo_token: ${{ secrets.SCORECARD_READ_TOKEN }}
+
+          # Publish the results for public repositories to enable scorecard badges. For more details, see
+          # https://github.com/ossf/scorecard-action#publishing-results.
+          # For private repositories, `publish_results` will automatically be set to `false`, regardless
+          # of the value entered here.
+          # publish_results: true
+
+      # Upload the results as artifacts (optional). Commenting out will disable uploads of run results in SARIF
+      # format to the repository Actions tab.
+      - name: "Upload artifact"
```

Comment:
Maybe we can disable the upload of the `sarif` as an artifact. As far as I understood, even if we don't upload the results as an artifact, it would still show the workflow results to us in the security tab, right? What is the purpose of uploading it?

Reply:
I think having this as an artifact would be fine. These artifacts are stored only for a limited time anyway and we don't treat them as something we need to preserve but can be handy from time to time for debugging. We do this for many jobs in CI in the main repo. Should not really impact running time much unless the report is gigantic :)

Reply:
Right. So this is ready to merge :D

Reply:
> Good point @r0qs, I forgot to remove it since it is optional, sorry for that (considering that you asked me for only the main features to be enabled)
> 
> Actually this do not interfere with the security tabs, that this Upload Artifact does is to uploads the scorecard run result in SARIF format as an artifact that anyone, (maintainer or user) can download and see. I will disable it commenting these lines for you to be able to enable in the future if you want.

No worries, we can keep it for now ;D

Reply:
Done

Reply:
I just saw the answer now 🥲, I just removed my last change haha